code/modules/mob/living/carbon/xenomorph/castes/warlock/abilities_warlock.dm

```diff
@@ -213,6 +213,14 @@
 	proj.iff_signal = null
 	frozen_projectiles += proj
 	take_damage(proj.damage, proj.ammo.damage_type, proj.ammo.armor_type, 0, turn(proj.dir, 180), proj.ammo.penetration)
+	if(obj_integrity <= 300)
```

Comment:
Switch statement? Also this could probably use percentages instead of flat numbers in case the integrity gets changed

Reply:
This is so bad lol for love of god use a %